chemie/shop/views.py

```diff
@@ -0,0 +1,381 @@
+            amount = form.cleaned_data.get("amount")
+            try:
+                receiver.profile.balance += amount
+                receiver.profile.save()
+            except InvalidOperation:
+                messages.add_message(
+                    request,
+                    messages.ERROR,
+                    (
+                        "Brukeren vil få mer enn 9999 HC-coins på "
+                        "konto om du fyller på med beløpet."
+                    ),
+                    extra_tags="Feil",
+                )
+                return render(
+                    request, "shop/refill-balance.html", {"form": form}
+                )
+            instance = form.save(commit=False)
+            instance.provider = provider
+            instance.save()
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Du har fylt på {} HC-coins til brukeren {}".format(
+                    amount, receiver.username
+                ),
+                extra_tags="Suksess",
+            )
+    return render(request, "shop/refill-balance.html", {"form": form})
+
+
+@permission_required("shop.add_item")
+def add_item(request):
+    form = AddItemForm(request.POST or None, request.FILES or None)
+    if request.POST:
+        if form.is_valid():
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Varen ble opprettet",
+                extra_tags="Suksess!",
+            )
+            form.save()
+            form = AddItemForm(None, None)
+    items = Item.objects.all()
+    context = {"form": form, "items": items}
+    return render(request, "shop/add_item.html", context)
+
+
+@permission_required("shop.edit_item")
+def edit_item(request, pk):
+    item = Item.objects.get(pk=pk)
+    form = EditItemForm(
+        request.POST or None, request.FILES or None, instance=item
+    )
+    if request.method == "POST":
+        if form.is_valid():
+            form.save()
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Varen ble endret.",
+                extra_tags="Endret!",
+            )
+            return redirect(reverse("shop:add-item"))
+    context = {"form": form}
+    return render(request, "shop/edit_item.html", context)
+
+
+@permission_required("shop.add_category")
+def add_category(request):
+    form = AddCategoryForm(request.POST or None)
+    if request.POST:
+        if form.is_valid():
+            form.save()
+    categories = Category.objects.all()
+    context = {"form": form, "categories": categories}
+    return render(request, "shop/add_category.html", context)
+
+
+@login_required
+def remove_item(request, pk):
+    item = Item.objects.get(pk=pk)
+    cart = ShoppingCart(request)
+    cart.remove(item)
+    return JsonResponse({"success": 1})
+
+
+def activate_happyhour(request):
```

Comment:
Trenger permission_required. Nå går det an for hvem som helst å aktivere Happy Hour med url-en .../shop/admin/activate-happyhour